kymatio/backend/base_backend.py

```diff
@@ -51,6 +56,8 @@ def fft_forward(self, x, direction='C2C', inverse=False):
                 output = self.ifft(x)
             else:
                 output = self.fft(x)
+        elif direction == 'R2C':
```

Comment:
It would make sens, here, to add a test for checking the input is indeed real.

Reply:
Yes.

Reply:
In one backend (tensorflow), we cast to complex before we send it into the fft. How do we reconcile that?

Reply:
> How do we reconcile that?
Reconcile what with what? The to this class (`FFT`) specifies that the input should be real, so we should check to make sure that it is. What it then does internally (such as for TF) is unrelated.

Reply:
The 'rfft' for tensorflow isn't the `tf.signal.rfft` but its `tf.sginal.fft`, and we cast the input to complex.

Reply:
Ok. So what's the problem?

Reply:
I was thinking that when it checks if the input tf tensor is real, that its gonna error out, b/c we cast to complex, but I'm guessing we could just make the tf not check for real?

Reply:
I still don't understand. In our wrapper, we check that the input is indeed real if `R2C` is specified, correct? Then, we may cast it to complex after that and pass it to `tf.signal.fft` and return the output. At what point are we providing a complex array where a real array is expected?

Reply:
Ok you are correct, I wasn't understanding the design, I get it now. 

Yeah, lets add the check for real.

Reply:
:+1: